### PR TITLE
Update various GcsObject cmdlets so they can be usable with Google Cloud Storage Provider.

### PR DESCRIPTION
**Pull request by quoctruong**
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/google-cloud-powershell/429)

### CODE REVIEW

File: Google.PowerShell/Storage/GcsObject.cs [hunk 1]
```diff
@@ -537,12 +578,38 @@ protected override void ProcessRecord()
...
 
+            // If we are in Google Cloud Storage Provider location, resolve the path to get possible bucket name and prefix.
+            string[] bucketAndPrefix = GetBucketAndPrefix();
+            string gcsProviderPrefix = null;
+            if (bucketAndPrefix != null && bucketAndPrefix.Length > 0)
+            {
+                Bucket = Bucket ?? bucketAndPrefix[0];
+                if (bucketAndPrefix.Length == 2)
+                {
+                    gcsProviderPrefix = bucketAndPrefix[1];
```
Comment (iantalarico) [inline Google.PowerShell/Storage/GcsObject.cs]: You use the same general logic above to get the prefix and bucket. Maybe make a shared function?

File: Google.PowerShell/Storage/GcsObject.cs [hunk 2]
```diff
@@ -588,16 +660,77 @@ protected override void ProcessRecord()
...
+            if (InputObject != null)
+            {
+                Bucket = InputObject.Bucket;
+                ObjectName = InputObject.Name;
+            }
+            else
+            {
+                // If we are in Google Cloud Storage Provider location, resolve the path to get possible bucket name and prefix.
+                string[] bucketAndPrefix = GetBucketAndPrefix();
+                if (bucketAndPrefix != null && bucketAndPrefix.Length > 0)
```
Comment (iantalarico) [inline Google.PowerShell/Storage/GcsObject.cs]: Same logic used here again

**Comment by quoctruong** [comment]
@iantalarico PTAL

File: Google.PowerShell.IntegrationTests/Storage/Storage.GcsObject.Tests.ps1 [hunk 3]
```diff
@@ -644,34 +701,99 @@ Describe "Get-GcsObject" {
...
-    It "should work" {
-        # Default ACLs set on the object from the bucket.
-        $obj = Get-GcsObject $bucket "testfile1.txt"
+    function Test-ObjectDefaultAcl($obj) {
         $obj.Acl.Count | Should Be 4
         $obj.Acl[0].ID.Contains("/project-owners-") | Should Be $true
         $obj.Acl[1].ID.Contains("/project-editors-") | Should Be $true
         $obj.Acl[2].ID.Contains("/project-viewers-") | Should Be $true
         $obj.Acl[3].ID.Contains("/user-") | Should Be $true
+    }
```
Comment (quoctruong) [inline Google.PowerShell.IntegrationTests/Storage/Storage.GcsObject.Tests.ps1]: We already checked that there are 4 ACLs so it shouldn't contain that.

**Comment by quoctruong** [comment]
Thanks!

**Pull request merged**